File: kotlin/src/main/kotlin/io/digitalstream/advanced/PatternMatching.kt
/**
 * Created by sebby on 9/25/16.
 */
package io.digitalstream.advanced

import io.digitalstream.advanced.Enums.Option

object PatternMatching {

  fun run(): Unit {
    val option = Option.Some("string")
    when (option) {
      is Option.Some -> println("got value: ${option.value}")
      is Option.None<*> -> println("got none")
    }
  }
}

File: scala/src/main/scala/io/digitalstream/advanced/PatternMatching.scala
/**
  * Created by sebby on 9/25/16.
  */
package io.digitalstream.advanced

object PatternMatching {

  def run(): Unit = {
    val option = Option("string")

    option match {
      case Some(value) => println(s"got value: $value")
      case None => println("got none")
    }
  }
}
